azurerm/internal/services/netapp/netapp_snapshot_resource.go

```diff
@@ -196,19 +198,42 @@ func resourceArmNetAppSnapshotDelete(d *schema.ResourceData, meta interface{}) e
 		return err
 	}
 
-	future, err := client.Delete(ctx, id.ResourceGroup, id.AccountName, id.PoolName, id.VolumeName, id.Name)
-	if err != nil {
-		if response.WasNotFound(future.Response()) {
-			return nil
-		}
+	if _, err = client.Delete(ctx, id.ResourceGroup, id.AccountName, id.PoolName, id.VolumeName, id.Name); err != nil {
 		return fmt.Errorf("Error deleting NetApp Snapshot %q (Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
 	}
 
-	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
-		if !response.WasNotFound(future.Response()) {
-			return fmt.Errorf("Error waiting for deleting NetApp Snapshot %q (Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
-		}
+	// The resource NetApp Snapshot depends on the resource NetApp Volume.
+	// Although the delete API returns 404 which means the NetApp Snapshot resource has been deleted.
+	// Then it tries to immediately delete NetApp Volume but it still throws error `Can not delete resource before nested resources are deleted.`
+	// In this case we're going to re-check status code again.
+	// For more details, see related Bug: https://github.com/Azure/azure-sdk-for-go/issues/11475
+	log.Printf("[DEBUG] Waiting for NetApp Snapshot Provisioning Service %q (Resource Group %q) to be deleted", id.Name, id.ResourceGroup)
```

Comment:
this isn't a Provisioning Service?

```suggestion
	log.Printf("[DEBUG] Waiting for NetApp Snapshot %q (Resource Group %q) to be deleted", id.Name, id.ResourceGroup)
```

Reply:
updated